### DISCUSSION TITLE
Is autolap possible?

### DISCUSSION THREAD

**Discussion by vdude69** [SauceLLC/sauce4zwift#33] [Ideas]
Hi,
sorry if it's a dumb question but I don't know if autolap (that is, taking laps automatically from a Zwift workout) can be possible.
I know laps can be created clicking on the relevant icon in the commands and that those icons can be set to be shown automatically and always on top in Sauce, but my need would be slightly different.
Let's say I have a workout like 10x 30" on/30" which I am riding on Zwift (or any other structured workout with lots of different laps)
It would be hard (at lesat for me) to go clicking on lap toggle every time (especially after the hard 30 secs intervals...lol) 
It would be great if we had a function that could relaunch lap data automatically, based on the workout laps I see on the screen and I do.
Let me know if this is technically possible, as said I'm totally new here.
Thanks

**Comment by mayfield** [comment]
Hi @vdude69,

Yes, this is possible.  There is a progress indicator in the data stream during workouts that represents how far into each interval you are.  It should be possible to monitor this and trigger laps based on it.  I've briefly outlined the enhancement ticket here: #30

**Reply by vdude69** [reply]
That's great news, thanks!
